python/nwm_client_new/src/hydrotools/nwm_client/NWMClient.py

```diff
@@ -0,0 +1,379 @@
+            temporary files.
+        dataframe_cache: ParquetCache, default ParquetCache("nwm_cache.parquet")
+            Local parquet directory used to locally cache retrieved dataframes.
+        catalog: NWMFileCatalog, optional, default GCPFileCatalog()
+            NWMFileCatalog object used to discover NWM files.
+        location_metadata_mapping: pandas.DataFrame with nwm_feature_id Index and
+            columns of corresponding site metadata. Defaults to 7500+ usgs_site_code
+            used by the NWM for data assimilation.
+        ssl_context: ssl.SSLContext, optional, default context
+            SSL configuration context.
+        cleanup_files: bool, default True
+            Delete downloaded NetCDF files upon program exit.
+
+        Returns
+        -------
+        NWMClient object
+        """
+        super().__init__()
+
+        # Set file output directory
+        self.file_directory = file_directory
+
+        # Set dataframe cache
+        if dataframe_cache:
+            self.dataframe_cache = dataframe_cache
+        else:
+            self._dataframe_cache = None
+
+        # Set file catalog
+        self.catalog = catalog
+
+        # Set crosswalk
+        self.crosswalk = location_metadata_mapping
+
+        # Set CA bundle
+        self.ssl_context = ssl_context
+
+        # Set cleanup flag
+        self.cleanup_files = cleanup_files
+
+    def get_cycle(
+        self,
+        configuration: str,
+        reference_time: str,
+        netcdf_dir: Union[str, Path]
+        ) -> dd.DataFrame:
+        """Retrieve a single National Water Model cycle as a 
+        dask.dataframe.DataFrame.
+        
+        Parameters
+        ----------
+        configuration: str, required
+            NWM configuration cycle.
+        reference_time: str, required
+            Reference time string in %Y%m%dT%HZ format. 
+            e.g. '20210912T01Z'
+        netcdf_dir: str or pathlib.Path, required
+            Directory to save downloaded NetCDF files.
+
+        Returns
+        -------
+        dask.dataframe.DataFrame of NWM data
+        """
+        # Generate list of urls
+        urls = self.catalog.list_blobs(
+            configuration=configuration,
+            reference_time=reference_time
+        )
+
+        # Setup downloader
+        downloader = FileDownloader(
+            output_directory=netcdf_dir,
+            create_directory=True,
+            ssl_context=self.ssl_context
+            )
+
+        # Download files
+        downloader.get(urls)
+
+        # Get dataset
+        ds = NWMFileProcessor.get_dataset(
+            input_directory=netcdf_dir,
+            feature_id_filter=self.crosswalk.index
+            )
+
+        # Convert to dataframe
+        df = NWMFileProcessor.convert_to_dask_dataframe(ds)
+
+        # Canonicalize
+        return NWMClient.canonicalize_dataframe(
+            df=df,
+            configuration=configuration
+        )
+
+    def get(
```

Comment:
It is not real clear (by reading the code at least) what the difference between `get_cycle` and `get` is, and it seems they are implemented in different ways.  Intuitively, I would think `get_cycle` would make multiple calls to `get`???  Also, why doesn't `get_cycle` attempt any caching or cleanup where `get` does?